src/tooltip.tsx

```diff
@@ -112,386 +161,442 @@ const Tooltip: React.FC<Props> = ({
-                            ? topAdjustment
-                            : window.pageYOffset + targetBoundingClientRect.current.top - distanceToTarget,
-
-                    left:
-                        targetBoundingClientRect.current.right + tooltipClientRect.width > window.innerWidth
-                            ? leftAdjustment
-                            : bottomAdjustment,
-                },
-                bottom: {
-                    top:
-                        targetBoundingClientRect.current.right + tooltipClientRect.width > window.innerWidth
-                            ? topAdjustment
-                            : window.pageYOffset + targetBoundingClientRect.current.bottom + distanceToTarget,
-                    left:
-                        targetBoundingClientRect.current.right + tooltipClientRect.width > window.innerWidth
-                            ? leftAdjustment
-                            : bottomAdjustment,
-                },
-            };
-
-            return containerPos[position];
-        },
-        [unstable_offsetX, tooltipClientRect]
-    );
+                break;
 
-    const getWidth = () => rest.width;
-    const width = getWidth();
-
-    const arrowClassNameByPosition = {
-        top: styles.arrowTop,
-        bottom: styles.arrowBottom,
-        left: styles.arrowLeft,
-        right: styles.arrowRight,
-    };
-
-    const vars =
-        position &&
-        assignInlineVars({
-            [styles.vars.enterTransform]: getEnterTransform(position),
-            [styles.vars.exitTransform]: getExitTransform(position),
-            [styles.vars.enterActiveAnimationName]: getEnterActiveAnimationName(position),
-            [styles.vars.enterDoneTransform]: getEnterDoneTransform(position),
-            [styles.vars.shadowAlpha]: getShadowAlpha(isDarkMode),
-            [styles.vars.arrowBoxShadow]: getArrowBoxShadow(position),
-        });
+            case 'left':
+                tooltipProps = {
+                    left: left - tooltipWidth - ARROW_SIZE / 2,
+                    top: Math.max(0, Math.min(maxTopOffset, (top + bottom - tooltipHeight) / 2)),
+                    padding: `0px ${TOOLTIP_PADDING_FROM_TARGET}px 0px 0px`,
+                };
 
-    React.useEffect(() => {
-        if (position && tooltipRef.current && isVisible) {
-            const widthAux = width ? width : 0;
-            setContainerPosition(getContainerPosition(position, widthAux));
+                break;
+
+            case 'right':
+            default:
+                tooltipProps = {
+                    left: right - TOOLTIP_OFFSET_FROM_TARGET,
+                    top: Math.max(0, Math.min(maxTopOffset, (top + bottom - tooltipHeight) / 2)),
+                    padding: `0px 0px 0px ${TOOLTIP_PADDING_FROM_TARGET}px`,
+                };
+
+                break;
         }
-    }, [isVisible, getContainerPosition, position, width]);
+
+        switch (finalPosition) {
```

Comment:
why 2 switch? I'd merge them into one

Reply:
this code went through many changes, but in this final version it's better to join the two switch blocks. Updated